proxy/src/auth/backend.rs

```diff
@@ -121,88 +110,173 @@ impl<'a, T, E> BackendType<'a, Result<T, E>> {
         use BackendType::*;
         match self {
             Console(c, x) => x.map(|x| Console(c, x)),
+            #[cfg(feature = "testing")]
             Postgres(c, x) => x.map(|x| Postgres(c, x)),
             Link(c) => Ok(Link(c)),
+            #[cfg(test)]
             Test(x) => Ok(Test(x)),
         }
     }
 }
 
-pub enum ComputeCredentials {
+pub struct ComputeCredentials<'a, T> {
+    pub info: ComputeUserInfo<'a>,
+    pub keys: T,
+}
+
+pub struct ComputeUserInfoNoEndpoint<'a> {
+    pub user: &'a str,
```

Comment:
I always was wondering. 
Is there any reason why it's `&str` and not just `String` or pointer.
That makes everything a way more complicated. Do you think it makes sense to replace it? Especially, because we already have `endpoint` and `cache_key` as `String`.

Reply:
I guess it was some prior attempt to avoid an allocation. But I agree, I'll change it